main.py

```diff
@@ -3,37 +3,93 @@
+class CustomFormatter(logging.Formatter):
+    format = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+
+    FORMATS = {
+        logging.DEBUG: Fore.WHITE + format + Style.RESET_ALL,
+        logging.INFO: Fore.CYAN + format + Style.RESET_ALL,
+        logging.WARNING: Fore.YELLOW + format + Style.RESET_ALL,
+        logging.ERROR: Fore.RED + format + Style.RESET_ALL,
+        logging.CRITICAL: Fore.RED + Style.BRIGHT + format + Style.RESET_ALL,
+    }
+
+    def format(self, record):
+        log_fmt = self.FORMATS.get(record.levelno, self.format)
+        formatter = logging.Formatter(log_fmt)
+        return formatter.format(record)
+
+# Configure the logger
+logger = logging.getLogger(__name__)
+handler = logging.StreamHandler(sys.stdout)
+handler.setFormatter(CustomFormatter())
+logger.addHandler(handler)
+logger.setLevel(logging.INFO)
 
-logging.basicConfig(level=logging.ERROR)
-class bot(commands.Bot):
+# Define a custom credits function
+def log_credits(message):
+    logger.info(Fore.BLACK + Back.WHITE + message + Style.RESET_ALL)
+
+class Bot(commands.Bot):
     def __init__(self):
-        intents=discord.Intents.all()
-        command_prefix='!'
+        intents = discord.Intents.all()
+        command_prefix = '!'
         super().__init__(command_prefix=command_prefix, intents=intents)
 
-        #get bot token from file
-        self.token = open("token.yml", "r").readline()
+        # Get bot token from file
+        try:
+            with open("token.yml", "r") as file:
+                self.token = file.readline().strip()
+        except FileNotFoundError:
+            logger.critical("Token file not found. Please ensure 'token.yml' exists.")
+            sys.exit(1)
+
         self.cogsfolder = "cogs"
 
-        # database
-        self.quote_db = sqlite3.connect("quote.db")
+        # Database connection
+        try:
+            self.quote_db = sqlite3.connect("quote.db")
+            logger.info("Connected to the SQLite database.")
+        except sqlite3.Error as e:
+            logger.critical(f"Database connection failed: {e}")
+            sys.exit(1)
 
-        #stuff that should not be reset after cog reload
+        # Persistent variables
         self.music_queue = {}
 
     async def load_extensions(self):
+        log_credits("EnhancedInit written by razrblds for Namaku")
+        log_credits("Currently using V1.5")
+        log_credits("This is made for YAMB and YAMB only. If you need a version for yourself, ask the contributor.")
+        logger.info("Loading cogs...")
+
+        tasks = []
         for file in os.listdir(self.cogsfolder):
-            if file.endswith(".py"): 
-                await self.load_extension(f"{self.cogsfolder}.{file[:-3]}")
+            if file.endswith(".py"):
+                cog_name = f"{self.cogsfolder}.{file[:-3]}"
+                tasks.append(self._load_cog(cog_name))
+
+        # Run tasks concurrently
+        await asyncio.gather(*tasks)
```

Comment:
this looks pretty unsafe in my opinion and cogs usually takes around less than a second to load

Reply:
yeah, i kinda wanted to make some "ultrafast" thing, you can remove that if needed